### DISCUSSION TITLE
Adding custom input field in autocompletion

### DISCUSSION THREAD

**Discussion by xinyuluo** [komoot/photon#665] [Q&A] [answered]
Hi! I'm trying to understand how autocomplete works in Photon. As far as I can see, photon uses edge ngram in elasticsearch to do autocompletion, but I am struggling to understand which part of Photon code takes care of this feature (except in [index_settings.json](https://github.com/komoot/photon/blob/master/es/index_settings.json#L72)), and is it possible to add my own input field in autocompletion?

Thank you.

**Comment by mtmail** [comment, marked as answer]
Is this question about edge ngrams or input fields? It's better create specific questions, ideally with examples.

**Reply by xinyuluo** [reply]
@mtmail one example of my use case is categorical search. Given that photon is a geocoder in nature it doesn't process general purpose queries like "restaurant in New York" or "pharmacies nearby" very well, and I want to handle this part of queries separately.
I was thinking if it's possible to allow queries like these to appear in autocompletion and handle them by some other search API if they were chosen from the suggestion list. Hence, I was interested in how autocompletion was implemented in Photon.
Does this make my question clearer?

**Reply by xinyuluo** [reply]
an example from [Qwant Maps](https://www.qwant.com/maps):
![image](https://user-images.githubusercontent.com/25223376/160626543-fe5fc959-1fc0-469a-8eb5-e494ff330f33.png)

**Reply by mtmail** [reply]
That sounds like it should be separate indeed and then results merged after querying Photon and the point-of-interest results. That's outside of Photon though. Likely any database with spatial features (be able to search inside a bounding box), like MySQL, Postgres+PostGIS, sqlite can be used for that purpose as no text-matching capabilities are needed.
